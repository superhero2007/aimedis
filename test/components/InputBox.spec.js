import React from 'react';
import { mount } from 'enzyme';

import InputBox from 'components/InputBox';

const email = (value) => {
  const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(value).toLowerCase());
};

function setup(label = 'Name or Company', height = 3.5,  validate = '', validationString = '') {
  return mount(<InputBox
    label={label}
    height={height}
    validate={validate}
    validationString={validationString}
  />);
}

describe('InputBox', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('change state properly', () => {
    const nameWrapper = setup('Name or Company', 3.5);
    expect(nameWrapper.state().flag).toEqual('BEGINNING');

    const nameInput = nameWrapper.find('.app__input__text');
    nameInput.simulate('blur');
    expect(nameWrapper.state().flag).toEqual('VALIDATE');

    const emailWrapper = setup('Name or Company', 3.5, email);
    expect(emailWrapper.state().flag).toEqual('BEGINNING');

    const emailInput = emailWrapper.find('.app__input__text');
    emailInput.simulate('blur');
    expect(emailWrapper.state().flag).toEqual('INVALIDATE');
  });
});
