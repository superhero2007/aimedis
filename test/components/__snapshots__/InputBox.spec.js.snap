// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should render properly 1`] = `"<div class=\\"app__input\\"><div class=\\"app__input__label\\">Name or Company</div><input type=\\"text\\" class=\\"app__input__text\\" style=\\"height: 3.5rem;\\"></div>"`;
